### PR TITLE
Add ordered and unordered lists components.

### LINKED ISSUE #363: Create ordered list and unordered list component
Create ordered list and unordered list component and add documentation. You can [find components on Zeplin](https://app.zeplin.io/styleguide/5d95b0aaad77af16c796a476/components?seid=5f4635fd373e6c284503ce1f).

--
Documentation
Ordered list

What does it do?
* Shows a list of related items in a specific order. List items are marked with numbers.
When to use it?
* When you want to show that items follow a specific order.
When not to use it?
* When the order of list items is not relevant. 
How to use it?
* List items should have similar line lengths so the one doesn’t overpower another.
* Start first list with 1, and second with 1.1, third 1.1.1 and fourth with 1.1.1.1.

Unordered list

What does it do?
* Shows a list of items in no particular order
When to use it?
* When the order of the items is not relevant. List items are marked with a bullet.
When not to use it?
* When order of list items is relevant and needs to follow a specific order.
How to use it?
* List items should have similar line lengths,  doesn’t overpower another.
* Include related items that are a similar level of importance. 

Relates to: https://github.com/wmcadigital/wmn-sprints/issues/21

### PR DESCRIPTION
**Pull request by catiarodriguescosta**
Fixes #363

### CODE REVIEW

**Review by daylesalmon — CHANGES_REQUESTED** [review]
Before I review and check any files...

There seems to be an issue with the SCSS changes you have made, as the preview: https://deploy-preview-371--wmnetwork.netlify.app/ does not load the styling correctly (it says 404 not found).

Usually this happens when the CSS hasn't been able to compile and move into the build file.
So just have another check in your SCSS changes for any mistakes...try starting the server locally and see if there are any errors reported in the terminal.

**Comment by catiarodriguescosta** [comment]
@daylesalmon should be fine now. :)

File: src/wmnds/components/lists/ordered-list/_ordered-list.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.wmnds-ordered-list {
+  counter-reset: item;
+  padding-inline-start: 0;
+
+  ol {
+    counter-reset: item;
+    padding-inline-start: $size-sm * 2;
```
Comment (daylesalmon) [inline src/wmnds/components/lists/ordered-list/_ordered-list.scss:7]:
```suggestion
    padding-inline-start: $size-lg;
```
I'd recommend changing this to the recommended. 
By multiplying by 2 we end up creating a new "size" which doesn't exist (1.5rem).

It's best to stick with our defined sizes.

File: src/wmnds/components/lists/unordered-list/_unordered-list.scss [hunk 2]
```diff
@@ -0,0 +1,18 @@
+.wmnds-unordered-list {
+  list-style-position: inside;
+  padding-inline-start: 0;
+
+  ul {
+    padding-inline-start: 2 * $size-sm;
```
Comment (daylesalmon) [inline src/wmnds/components/lists/unordered-list/_unordered-list.scss:6]:
```suggestion
    padding-inline-start: $size-lg;
```

Same as previous comment with sizing.

File: src/www/pages/components/lists/index.njk [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+              <ol>
+                <li>Text</li>
+              </ol>
+            </li>
+          </ol>
+        </li>
+      </ol>
+  </li>"
+      }
+    ) | trim
```
Comment (daylesalmon) [inline src/www/pages/components/lists/index.njk:48]: Indent, so it is a bit easier to read.

File: src/www/pages/components/lists/index.njk [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+              <li>Text
+                <ul>
+                  <li>Text</li>
+                </ul>
+              </li>
+            </ul>
+          </li>
+      </ul>
+  </li>"
+        }
```
Comment (daylesalmon) [inline src/www/pages/components/lists/index.njk:95]: Indent, so it is easier to read.

File: src/www/pages/components/lists/index.njk [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            </li>
+          </ol>
+        </li>
+      </ol>
+  </li>"
+      }
+    ) | trim
+  ])
+}}
+
```
Comment (daylesalmon) [inline src/www/pages/components/lists/index.njk:51]:
```suggestion
<br/><br/>

```

This makes spacing consistent with other preview pages, i.e. https://deploy-preview-371--wmnetwork.netlify.app/components/inset-text/

**Review by daylesalmon — CHANGES_REQUESTED** [review]
Looks good, just a few minor changes.

File: src/www/pages/components/lists/index.njk (hunk 3 above)
Comment (catiarodriguescosta) [reply src/www/pages/components/lists/index.njk:48]: Sorry, I know, but it was the only way I found to have it right formatted on the code snippet. I know it is ugly here but it is perfectly indented on the code snippet.

**Review by gldgrnt — COMMENTED** [review]
@daylesalmon & @catiarodriguescosta  I don't know how much of an edge case this is, but nesting a `wmnds-unordered-list` inside of a `wmnds-ordered-list` one does this:

image

Instead of producing bullet points

**Comment by catiarodriguescosta** [comment]
@gldgrnt it should be fixed now. Could you try again? Wait for the new preview link.

**Comment by gldgrnt** [comment]
That fixes the indentation, but the nested unordered list shows numbers instead of bullet points.

**Comment by catiarodriguescosta** [comment]
@daylesalmon do you think this page needs a mixed list example? I didn't explore a lot this option.

**Comment by KaterinaKir** [comment]
No, we are not going to mix ordered lists with unordered. We want content editors to choose one or the other based on user need.

**Comment by gldgrnt** [comment]
Cool!

**Review by gldgrnt — APPROVED** [review]
All good with me

**Review by daylesalmon — CHANGES_REQUESTED** [review]
@catiarodriguescosta 

We need this to work in almost all possible variants ul>li>ul, ol>li>ul, ol>li>ol>li>ul etc.
Update and test the various different ways this can be used.

You should include the different ways on the example page so users can see the different methods of nesting this and how it will look.

Here is some sample code of what I did on the [Swift's T+C's page](https://www.networkwestmidlands.com/terms-conditions/swift-card):

````css
ul li:before {
  content: "\2022\ ";
  color: #3c1053;
  vertical-align: middle;
  font-size: 32px;
  font-size: 3.2rem;
  line-height: 36px;
  line-height: 3.6rem;
}
ul ul li:before {
  color: #9d5baf;
}
ol:not([type="a"]) {
  counter-reset: nwm-counter;
}
ol[type="a"] {
  counter-reset: nwm-letter-counter;
}
ol li:before {
  content: counters(nwm-counter, ".") ".";
  counter-increment: nwm-counter;
  color: #9d5baf;
  font-weight: bold;
  margin-right: 10px;
}
ol[type="a"] > li:before {
  content: counter(nwm-letter-counter, lower-alpha) ". ";
  counter-increment: nwm-letter-counter;
}
ol li ol {
  padding-left: 20px;
}

**Comment by daylesalmon** [comment]
@catiarodriguescosta 

image
In the above screenshot, the bottom list should show 1.2. not 1.1. 

This happens when using the below code:
image


When this is fixed it may be worth updating the example to show a more complicated example like such.

File: src/wmnds/components/lists/ordered-list/_ordered-list.scss [hunk 6]
```diff
@@ -0,0 +1,21 @@
+.wmnds-ordered-list {
```
Comment (daylesalmon) [inline src/wmnds/components/lists/ordered-list/_ordered-list.scss:1]:
```suggestion
.wmnds-ordered-list, ol {
```

File: src/wmnds/components/lists/unordered-list/_unordered-list.scss [hunk 7]
```diff
@@ -0,0 +1,20 @@
+.wmnds-unordered-list {
```
Comment (daylesalmon) [inline src/wmnds/components/lists/unordered-list/_unordered-list.scss:1]:
```suggestion
.wmnds-unordered-list, ul {
```

**Review by daylesalmon — CHANGES_REQUESTED** [review]

File: src/wmnds/components/lists/ordered-list/_ordered-list.scss [hunk 8]
```diff
@@ -0,0 +1,21 @@
+.wmnds-ordered-list, ol {
```
Comment (gldgrnt) [inline src/wmnds/components/lists/ordered-list/_ordered-list.scss:1]: You might have to remove the `ol` from this line

File: src/wmnds/components/lists/unordered-list/_unordered-list.scss [hunk 9]
```diff
@@ -0,0 +1,20 @@
+.wmnds-unordered-list, ul {
```
Comment (gldgrnt) [inline src/wmnds/components/lists/unordered-list/_unordered-list.scss:1]: You might have to remove the `ul` from this line

**Review by gldgrnt — CHANGES_REQUESTED** [review]
Targeting `<ul>` and `<ol>` sitewide is causing issues:

image

File: src/wmnds/components/lists/unordered-list/_unordered-list.scss (hunk 9 above)
Comment (catiarodriguescosta) [reply src/wmnds/components/lists/unordered-list/_unordered-list.scss:1]: Done!

File: src/wmnds/components/lists/ordered-list/_ordered-list.scss (hunk 8 above)
Comment (catiarodriguescosta) [reply src/wmnds/components/lists/ordered-list/_ordered-list.scss:1]: Done

**Review by gldgrnt — APPROVED** [review]
Looks good to me

**Comment by daylesalmon** [comment]
I've updated the lists so that they now work correctly on global ul/ol elements. This will allow content editors in future to be able to use the correct list styles if they don't have access to the code.

**Review by daylesalmon — APPROVED** [review]

**Review by gldgrnt — APPROVED** [review]
I like those additions!

**Pull request merged**